yocs_diff_drive_pose_controller/src/diff_drive_pose_controller_ros.cpp

```diff
@@ -248,4 +254,22 @@ void DiffDrivePoseControllerROS::disableCB(const std_msgs::EmptyConstPtr msg)
   }
 }
 
+void DiffDrivePoseControllerROS::reconfigCB(yocs_msgs::PoseControllerConfig &config, uint32_t level)
+{
+  dynamic_reconfig_mutex_.lock();
```

Comment:
This is the wrong mutex, you have to lock with the same mutex as where you use the variables :/